lib/tests/group-strategy.test.ts

```diff
@@ -565,10 +567,20 @@ test("group-strategy", async () => {
         },
       ]);
 
+      const endpoints = {
```

Comment:
I'd need to benchmark the `endpoints` object to give you a good answer here, but I'd expect similar issues to that of the `schemas` object.

Actually on this topic I had an idea that needs testing but might work. If Typescript applies the same optimisations to namespace modules as regular modules, then each section of the template (i.e. schemas, endpoints) could be contained by a namespace module, allowing for improved performance with basically the same interface being defined (also means the file grouped strategies can share an implementation, difference between them being one uses inlined namespace modules and the other separate files).

Tangentially related to the above, would you be alright if [this check](https://github.com/astahmer/openapi-zod-client/blob/3933722e0c6d7a7b7af50dd63d31aa5f91f666a1/lib/src/template-context.ts#LL98C29-L98C29) is yeeted. If endpoints are grouped under `default` (as is used for ungrouped endpoints in a grouped context, technically you could use the empty string for full backwards compatibility but I'd advise against it) in all situations rather than this special behaviour, then the `group.hbs` and `default.hbs` templates can reduced to *just* a single default template which greatly simplifies the libraries implementation.

Also what're your thoughts regarding making the default template just infer its schema types, haven't noticed any costs associated with this outside of a large line count, and it's almost always something you'll want to do as a consumer of the library anyway (as your applications types should flow from the API boundary directly) so it seems like it adds pointless boilerplate to not have it by default.

Reply:
> Also what're your thoughts regarding making the default template just infer its schema types, haven't noticed any costs associated with this outside of a large line count, and it's almost always something you'll want to do as a consumer of the library anyway (as your applications types should flow from the API boundary directly) so it seems like it adds pointless boilerplate to not have it by default.

if that doesn't slow TS, I don't see any downside there, feel free to add this as well

Reply:
> > In general if you're comfortable with some slight breaking changes (e.g. a default prefix by default) to the default template, then you can improve maintainability and performance. Avoided making changes like this before, but if you're fine with 'em I'll go ahead with them.
> 
> I am ! this PR already adds some breaking changes so, it feels a great timing if you want to add some more

Will do. You alright if I continue to overload this PR, rather than splitting it out into changesets? Would prefer to do so as many of the changes are dependant on each other, so it'd be less work on my end.

Reply:
yes, please go for what's easiest for you to work with